src/main/java/server/router/RequestRouter.java

```diff
@@ -1,54 +1,41 @@
 package server.router;
 
-import java.util.List;
-import server.constants.Protocol;
+import java.util.HashMap;
+import server.constants.Path;
 import server.constants.Status;
 import server.request.Request;
 import server.response.Response;
-import server.response.ResponseBuilder;
 import server.router.routes.HeadRequest;
-import server.router.routes.Route;
+import server.router.routes.RouteHandler;
 import server.router.routes.SimpleGet;
 import server.router.routes.SimpleGetWithBody;
 
 public class RequestRouter {
-  private Request request;
-
   public Response getResponse(Request request) {
-    this.request = request;
-    List<Route> routes = getAllRoutes();
+    HashMap<Path, RouteHandler> routes = getAllRoutes();
+    RouteHandler route = routes.get(request.path());
 
     if (request.path() == null) {
```

Comment:
This check can be unnecessary. If we didn't find a RouteHandler then method `getResponse` can  return `new Response(Status.NOT_FOUND.format(), null, null)` instead of null. What do you think?